applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -67,7 +67,7 @@ def AnalyzeDesignAndReportToCommunicator( self, currentDesign, optimizationItera
             # response gradients
             if communicator.isRequestingGradientOf(identifier):
                 response.CalculateGradient()
-                communicator.reportGradient(identifier, response.GetShapeGradient())
+                communicator.reportGradient(identifier, response.GetNodalGradient(KM.SHAPE_SENSITIVITY))

```

Comment:
What is the motivation to set the variable explicitly here? Isn't it more general if the variable that contains the sensitivity is not explicity set by the analyzer?

Reply:
The goal of this PR is to generalize the response interface, so the non generic function `GetShapeGradient` was generalized to `GetNodalGradient`. 
This now requires to specify the variable at the calling side, e.g. the optimization. The **Shape**OptimizationApplication only deals with `SHAPE` variables - and currently there are no plans to use other variables.